common/ledger/blkstorage/blockindex.go

```diff
@@ -267,6 +305,28 @@ func constructTxIDKey(txID string, blkNum, txNum uint64) []byte {
 	return append(k, util.EncodeOrderPreservingVarUint64(txNum)...)
 }
 
+// decodeTxIDKey takes input an encoded txid key of the format `prefix:len(TxID):TxID:BlkNum:TxNum`
+// and returns the bytes segment `len(TxID):TxID` from this
+func decodeTxIDKey(encodedTxIDKey []byte) ([]byte, error) {
+	if len(encodedTxIDKey) == 0 {
+		return nil, errors.New("Invalid txIDKey - zero-length slice")
+	}
+	if encodedTxIDKey[0] != txIDIdxKeyPrefix {
+		return nil, errors.Errorf("Invalid txIDKey {%x} - unexpected prefix", encodedTxIDKey)
+	}
+	remainingBytes := encodedTxIDKey[utf8.RuneLen(txIDIdxKeyPrefix):]
+
+	txIDLen, n, err := util.DecodeOrderPreservingVarUint64(remainingBytes)
+	if err != nil {
+		return nil, errors.WithMessagef(err, "Invalid txIDKey {%x}", encodedTxIDKey)
+	}
+	remainingBytes = remainingBytes[n:]
+	if len(remainingBytes) <= int(txIDLen) {
+		return nil, errors.Errorf("Invalid txIDKey {%x}, fewer bytes present", encodedTxIDKey)
+	}
+	return encodedTxIDKey[1 : n+int(txIDLen)+1], nil
```

Comment:
To retain consistent usage within this function:
```suggestion
	return encodedTxIDKey[utf8.RuneLen(txIDIdxKeyPrefix): n+int(txIDLen)+1], nil
```

If we decide to compute the length of txID at the caller, this can be simplified to 
```suggestion
	return encodedTxIDKey[txIDStart : txIDEnd+1], nil
```

Reply:
yes, agreed.